### PR TITLE
feat(textproto): allow compilation units to contain multiple sources

### PR DESCRIPTION
**Pull request by justbuchanan**
Some portions of google3 contain so many textproto build targets that it is causing performance problems for blaze. Stuffing multiple textprotos in a single compilation unit will allow us to reduce the number of build targets.

### CODE REVIEW

File: kythe/cxx/indexer/textproto/analyzer.cc [hunk 1]
```diff
@@ -838,16 +838,19 @@ absl::Status AnalyzeCompilationUnit(PluginLoadCallback plugin_loader,
                                     const proto::CompilationUnit& unit,
                                     const std::vector<proto::FileData>& files,
                                     KytheGraphRecorder* recorder) {
-  if (unit.source_file().size() != 1) {
+  if (unit.source_file().size() > 1) {
```
Comment (shahms) [inline kythe/cxx/indexer/textproto/analyzer.cc:841]: `< 1` or `unit.source_file().empty()`

File: kythe/cxx/indexer/textproto/analyzer.cc [hunk 2]
```diff
@@ -838,16 +838,19 @@ absl::Status AnalyzeCompilationUnit(PluginLoadCallback plugin_loader,
...
-        "Expected Unit to contain 1 source file");
+        "Expected Unit to contain 1+ source files");
   }
   if (files.size() < 2) {
     return absl::FailedPreconditionError(
         "Must provide at least 2 files: a textproto and 1+ .proto files");
   }
 
-  const std::string textproto_name = unit.source_file(0);
+  std::set<std::string> textproto_filenames;
```
Comment (shahms) [inline kythe/cxx/indexer/textproto/analyzer.cc:850]: `absl::flat_hash_set<std::string>` (unless you need order or pointer stability)

File: kythe/cxx/indexer/textproto/analyzer.cc [hunk 3]
```diff
@@ -868,15 +871,17 @@ absl::Status AnalyzeCompilationUnit(PluginLoadCallback plugin_loader,
   }
   LOG(INFO) << "Proto message name: " << message_name;
 
+  std::map<std::string, const proto::FileData*> file_data_by_path;
```
Comment (shahms) [inline kythe/cxx/indexer/textproto/analyzer.cc:874]: absl::flat_hash_map

**Review by shahms — CHANGES_REQUESTED** [review]

File: kythe/cxx/indexer/textproto/analyzer.cc (hunk 1 above)
Comment (justbuchanan) [reply kythe/cxx/indexer/textproto/analyzer.cc:841]: Oops, good catch!

**Review by shahms — APPROVED** [review]

**Pull request merged**